features/largeVideo/components/LargeVideo.js

```diff
@@ -63,9 +64,14 @@ class LargeVideo extends Component {
             this.props.dispatch(participantSelected(activeParticipant.id));
         }
 
+        let mirrored = videoTrack &&
+            videoTrack.isLocal() &&
+            videoTrack.videoType === 'camera';
```

Comment:
Regarding the literal, it looks like we should use the track.isScreenSharing() method from lib-jitsi-meet instead but it appears unimplemented.

https://github.com/jitsi/lib-jitsi-meet/blob/master/modules/RTC/JitsiTrack.js#L236

I'd even go so far as to add a track.shouldShowMirrored() method to JitsiTrack.

I'll give the other issues some thought for tomorrow.